### PR TITLE
Implementation to verify categories in request

### LINKED ISSUE #102: Verify categories in request
The API should only accept categories (device categories, consumable categories, personnel qualifications, personnel areas) that are defined in the configuration for the given region.

### PR DESCRIPTION
**Pull request by MaxWies**
Categories of languages are verified using the region code.

Methods only used in ResourceController.cs and I am still thinking if the verfication task should be done by CondigurationService.cs

Added Unit tests.

I still have to test them with the api.

### CODE REVIEW

**Comment by chaoran-chen** [comment]
This branch can be tested at https://test8.pirat.chaoran-chen.de/, new commits to this branch will be automatically deployed. You can see the current version at the bottom-left.

**Comment by chaoran-chen** [comment]
When I try to create an offer with a real-time PCR Thermocycler, I am getting a 400 Bad Request with the message `"1006:InvalidCategoryDevice"`

The device object:
```
    {
      "category": "RT_PCR_THERMOCYCLER",
      "name": "3",
      "manufacturer": "2",
      "ordernumber": "1",
      "amount": 34,
      "annotation": ""
    }
```

**Comment by MaxWies** [comment]
> 
> 
> When I try to create an offer with a real-time PCR Thermocycler, I am getting a 400 Bad Request with the message `"1006:InvalidCategoryDevice"`
> 
> The device object:
> 
> ```
>     {
>       "category": "RT_PCR_THERMOCYCLER",
>       "name": "3",
>       "manufacturer": "2",
>       "ordernumber": "1",
>       "amount": 34,
>       "annotation": ""
>     }
> ```

The frontend already converts the category name that is seen by the user into the uppercase german/english word that is used as key and as value to store in the db?

Things to know earlier...

**Comment by chaoran-chen** [comment]
This is what the keys are for, keys are the language in which the services communicate^^ The backend does not even know which natural language is used (only the region).

**Comment by MaxWies** [comment]
Should be mergable

File: Pirat/Services/ConfigurationService.cs [hunk 1]
```diff
@@ -103,11 +105,60 @@ public List<string> GetRegionCodes()
...
+        {
+            if(!Regions[regionCode].Categories.Device.Contains(category))
+            {
+                throw new ArgumentException(Codes.FailureCodes.InvalidCategoryDevice);
+            }
+        }
+
+        public void ThrowIfNotPersonnelAreaInLanguage(string regionCode, List<string> areas)
+        {
+            if (!areas.Any() || areas.Any(area => !Regions[regionCode].Categories.PersonnelArea.Contains(area)))
```
Comment (chaoran-chen) [inline Pirat/Services/ConfigurationService.cs:146]: Why is it throwing an error if `areas` is empty? Then, every provided area (which is none) would be defined.

File: Pirat/Services/ConfigurationService.cs [hunk 2]
```diff
@@ -69,9 +71,9 @@ public RegionClientConfig GetConfigForRegion(string regionCode)
             }
             else
             {
-                using var scope = _logger.BeginScope("Unknown region-code");
+                using var scope = _logger.BeginScope("Unknown region-regionCode");
```
Comment (chaoran-chen) [inline Pirat/Services/ConfigurationService.cs]: `region-regionCode`?

File: Pirat/Services/IConfigurationService.cs [hunk 3]
```diff
@@ -39,6 +40,16 @@ public interface IConfigurationService
         /// </returns>
         List<string> GetRegionCodes();
 
+        void ThrowIfNotInLanguage(string regionCode, Offer offer);
```
Comment (chaoran-chen) [inline Pirat/Services/IConfigurationService.cs]: `NotInRegion` instead of `NotInLanguage`?

File: Pirat/Services/ConfigurationService.cs (hunk 1 above)
Comment (MaxWies) [reply Pirat/Services/ConfigurationService.cs:146]: Now it throws an error if none, empty or an element is not a valid value

File: Pirat/Services/IConfigurationService.cs [hunk 4]
```diff
@@ -39,6 +40,16 @@ public interface IConfigurationService
...
 
+        void ThrowIfNotInRegion(string regionCode, Offer offer);
+
+        void ThrowIfNotConsumableCategoryInLanguage(string regionCode, string category);
+
+        void ThrowIfNotDeviceCategoryInLanguage(string regionCode, string category);
+
+        void ThrowIfNotPersonnelAreaInLanguage(string regionCode, List<string> areas);
+
+        void ThrowIfNotPersonnelQualificationInLanguage(string regionCode, List<string> qualifications);
```
Comment (chaoran-chen) [inline Pirat/Services/IConfigurationService.cs]: `InLanguage -> InRegion` - If you change it once, you should probably change it everywhere^^

**Review by chaoran-chen — APPROVED** [review]

**Pull request merged**